enterprise/suppress-user/factory.go

```diff
@@ -26,57 +31,197 @@ func (m *Factory) Setup(ctx context.Context, backendConfig backendconfig.Backend
 	if m.Log == nil {
 		m.Log = logger.NewLogger().Child("enterprise").Child("suppress-user")
 	}
+
 	if m.EnterpriseToken == "" {
 		m.Log.Info("Suppress User feature is enterprise only")
 		return &NOOP{}, nil
 	}
+
 	m.Log.Info("Setting up Suppress User Feature")
+
 	backendConfig.WaitForConfig(ctx)
-	var repository Repository
-	if config.GetBool("BackendConfig.Regulations.useBadgerDB", true) {
-		tmpDir, err := misc.CreateTMPDIR()
+
+	var pollInterval time.Duration
+	config.RegisterDurationConfigVariable(300, &pollInterval, true, time.Second, "BackendConfig.Regulations.pollInterval")
+
+	useBadgerDB := config.GetBool("BackendConfig.Regulations.useBadgerDB", true)
+	if useBadgerDB {
+		identity := backendConfig.Identity()
+
+		fullSuppressionPath, latestSuppressionPath, err := getRepoPath()
 		if err != nil {
-			return nil, fmt.Errorf("could not create tmp dir: %w", err)
+			return nil, fmt.Errorf("could not get repo path: %w", err)
 		}
-		path := path.Join(tmpDir, "suppression")
 
-		// TODO: implement seeder source, to retrieve the initial state from a persisted backup
-		var seederSource func() (io.Reader, error)
+		if !alreadySeeded(fullSuppressionPath) && config.IsSet("SUPPRESS_USER_BACKUP_SERVICE_URL") {
+			repo := &RepoSwitcher{
+				mu: sync.RWMutex{},
+			}
```

Comment:
we may create the repo after we have created the latestRepo, so that we can simply do everything in a single line

```go
repo := &RepoSwitcher{Repository: latestRepo}
```